cmd/debos/debos.go

```diff
@@ -214,7 +223,13 @@ func main() {
 	}
 
 	if !options.DisableFakeMachine && !fakemachine.InMachine() && fakemachine.Supported() {
```

Comment:
This means that if the user explicitly decided to use a fakemachine backend (e.g. kvm) debos will still fall back on the non-fakemachine case if it's not supported.

Reply:
thanks, i've sorted out that case, now debos will not fallback to the non-fakemachine case if a backend is explicitly set.

Reply:
Yes but if fakemachine is not supported at all it will *still* fallback :) as fakemachine.Supported() returns false

And tbh these conditions are getting out of hand :). You probably want to determine what to launch earier so the remainder of the flow is easier.. Fwiw the code is a bit tricky as it needs to match the following cases:

* outer debos
  - If backend is "auto", use fakemachine if supported otherwise fallback to non-fakemachine
 - if backend is set to something else force fakemachine usage and fail if it doesn't work

* inner debos:
 - don't ever start fakemachine and start the actions; (this is determed by the InMachine() check)

Fwiw as test case you probably should have:
- normal setup:
  * automatic usage of (a) fakemachine (backend)
  * automatic fallback to non-fakemachine 
- disabled fakemachine usage
 * automatic usage of non fakemachine (no backends option)
 * failure when specifying a backend

Reply:
OK i agree, it's caused me a lot of headscratching each time i read the code :-)

hopefully now i have cleaned that up a little bit...